src/other/index.jl

```diff
@@ -128,7 +128,7 @@ function Base.empty!(x::Index)
 end
 
 function Base.insert!(x::Index, idx::Integer, nm::Symbol)
-    1 <= idx <= length(x.names)+1 || error(BoundsError())
+    1 <= idx <= length(x.names)+1 || throw(BoundsError("attempt to insert to a data frame with $(length(x.names)) columns at index $idx"))
```

Comment:
```suggestion
    if !(1 <= idx <= length(x.names)+1)
    throw(BoundsError("attempt to insert to a data frame with $(length(x.names)) columns at index $idx"))
     end
```

Reply:
Looks good to me, but @nalimilan has a better feel for language issues 😄. Thank you.

Reply:
Maybe this?
```suggestion
    if !(1 <= idx <= length(x.names)+1)
        throw(BoundsError("attempt to insert a column to a data frame with $(length(x.names)) columns at index $idx"))
     end
```